save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/CloneRepositoryController.kt

```diff
@@ -45,24 +44,18 @@ class CloneRepositoryController(
      */
     @PostMapping(value = ["/submitExecutionRequest"])
     fun submitExecutionRequest(@RequestBody executionRequest: ExecutionRequest): Mono<StringResponse> {
-        val project = executionRequest.project
-        val projectId: Long
-        try {
-            projectId = projectService.saveProject(project)
-            executionRequest.project.id = projectId
-            log.info("Project $projectId saved")
-        } catch (exception: DataAccessException) {
-            log.error("Error when saving project for execution request $executionRequest", exception)
-            return Mono.just(ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error to clone repo"))
-        }
-        log.info("Sending request to preprocessor to start cloning project id=$projectId")
-        return preprocessorWebClient
-            .post()
-            .uri("/upload")
-            .body(Mono.just(executionRequest), ExecutionRequest::class.java)
-            .retrieve()
-            .toEntity(String::class.java)
-            .toMono()
+        val projectExecution = executionRequest.project
+        val project = projectService.getProjectByNameAndOwner(projectExecution.name, projectExecution.owner)
+        project?.let {
+            log.info("Sending request to preprocessor to start cloning project id=${it.id}")
+            return preprocessorWebClient
+                .post()
+                .uri("/upload")
+                .body(Mono.just(executionRequest), ExecutionRequest::class.java)
+                .retrieve()
+                .toEntity(String::class.java)
+                .toMono()
```

Comment:
`bodyToMono`?

Reply:
But there should be `Mono<StringResponse>`, not `Mono<String>`

Reply:
Ok